### PR TITLE
Feature/tech 8096

### PR DESCRIPTION
**Pull request by Sameer-Begh**

### CODE REVIEW

File: src/components/customCarousel/index.tsx [hunk 1]
```diff
@@ -21,6 +21,27 @@ interface ICustomCarouselProps {
...
+      if (currentIndex === dummyImage?.length - 1) {
+        setIsLeftDisabled(false);
+        setIsRightDisabled(true);
+        return;
+      }
+      setIsLeftDisabled(false);
+      setIsRightDisabled(false);
+    };
+    setDisableStatus();
+  }, [currentIndex]);
```
Comment (shelly19gkmit) [inline src/components/customCarousel/index.tsx:43]: Can be refactored more using a single object state instead creating individual states-
const [disabledState, setDisabledState] = useState({
  isRightDisabled: false,
  isLeftDisabled: false,
});

useEffect(() => {
  setDisabledState(prevState => ({
    ...prevState,
    isLeftDisabled: currentIndex === 0,
    isRightDisabled: currentIndex === dummyImage?.length - 1,
  }));
}, [currentIndex, dummyImage]);

Plz do a sanity after refactoring this and add conditions based on requirement

File: src/components/customCarousel/index.tsx [hunk 2]
```diff
@@ -38,27 +59,24 @@ const CustomCarousel: FC<ICustomCarouselProps> = props => {
...
-              src={CAROUSEL}
-              alt={'Carousel_left_icon'}
-              height={16}
-              width={10.29}
-            />
-          </Box>
-        </Button>
+      <Box
+        onClick={isDisable ? undefined : onClick}
+        className={`${isDisable ? 'none' : 'cursor-pointer select-none'} ${(isTransformLeft && !isLeftDisabled) || (isTransformRight && isRightDisabled) ? 'scale-x-[-1]' : ''}`}>
```
Comment (shelly19gkmit) [inline src/components/customCarousel/index.tsx:70]: Conditions should be defined before return statement

File: src/components/ui/button/index.tsx [hunk 3]
```diff
@@ -1,65 +1,161 @@
...
+  const renderButtonBorder = () => {
+    return (
+      <>
+        <path
+          d={getSVGButtonBorder[width] || ''}
+          fill={borderColor}
+          mask="url(#path-1-inside-1_102_9628)"
+        />
+      </>
+    );
```
Comment (shelly19gkmit) [inline src/components/ui/button/index.tsx:93]: const renderButtonBorder = () => {
    return (
        <path
          d={getSVGButtonBorder[width] || ''}
          fill={borderColor}
          mask="url(#path-1-inside-1_102_9628)"
        />
    );
}

File: src/components/ui/button/index.tsx [hunk 4]
```diff
@@ -1,65 +1,161 @@
...
+        x={!isIcon ? '50%' : '54.5%'}
+        y={!isIcon ? '50%' : '54%'}
+        alignment-baseline="middle"
+        text-anchor="middle"
+        className="text-center text-middle tracking-wider"
+        fill={color}
+        font-size={fontSize}
+        fontWeight={fontWeight}>
+        {label}
+      </text>
```
Comment (shelly19gkmit) [inline src/components/ui/button/index.tsx:80]: Maintain reusable text component

File: src/components/customCarousel/index.tsx (hunk 1 above)
Comment (Sameer-Begh) [reply src/components/customCarousel/index.tsx]: done

File: src/components/campaignbanner/index.tsx [hunk 5]
```diff
@@ -53,7 +53,13 @@ export const CampaingBanner: FC<ICampaingBanner> = props => {
   const renderCarouseWithStartBtn = () => {
     return (
```
Comment (mandeepatindigg) [inline src/components/campaignbanner/index.tsx:54]: @Sameer-Begh file name is not camelCase

**Pull request merged**